rclcpp_lifecycle/test/test_lifecycle_service_client.cpp

```diff
@@ -355,3 +356,55 @@ TEST_F(TestLifecycleServiceClient, lifecycle_transitions) {
   transitions = lifecycle_client()->get_available_transitions();
   EXPECT_EQ(transitions.size(), 0u);
 }
+
+
+TEST_F(TestLifecycleServiceClient, get_service_names_and_types_by_node)
+{
+  auto node1 = std::make_shared<LifecycleServiceClient>("client1");
+  auto node2 = std::make_shared<LifecycleServiceClient>("client2");
+
+  const auto * node_graph =
+    dynamic_cast<rclcpp::node_interfaces::NodeGraph *>(node1->get_node_graph_interface().get());
+  ASSERT_NE(nullptr, node_graph);
+
+  EXPECT_THROW(
```

Comment:
You can be the first user of my new testing macros! I've added `RCLCPP_EXPECT_THROW_EQ` and `RCLCPP_ASSERT_THROW_EQ`, in https://github.com/ros2/rclcpp/blob/master/rclcpp/test/utils/rclcpp_gtest_macros.hpp.

It has the benefit of not only checking that this throws a runtime error, but that the message contained in the runtime error matches the expected. It's also implemented for RCLError types. Obviously, only use these versions when the contents of the exception can be depended on.

See example uses here:
https://github.com/ros2/rclcpp/blob/master/rclcpp/test/utils/test_rclcpp_gtest_macros.cpp#L42-L56

Reply:
This test is in `rclcpp_lifecycle`, I can't include the header `rclcpp_gtest_macros.hpp`. Did you expose this macro/header to other packages ?

Reply:
Oh you're right. Oh well, we can wait until it becomes more mainsteam.